File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03651/p03651_num6_parsed.py
from typing import List

def snuke_possible(n: int, k: int, a: List[int]) -> str:
    # Check if K is already present in the balls
    if k in a:
        return 'POSSIBLE'
    
    # Check if it is possible to reach K using the operations
    max_ai = max(a)
    if k > max_ai:
        return 'IMPOSSIBLE'
    
    # Check if the sum of all integers is greater than or equal to K
    sum_ai = sum(a)
    if k > sum_ai:
        return 'IMPOSSIBLE'
    
    # Check if the sum of all integers and K is odd, and if the maximum integer is odd
    if (sum_ai + k) % 2 != 0 or max_ai % 2 != 0:
        return 'IMPOSSIBLE'
    
    return 'POSSIBLE'

# Read the input from standard input
n, k = map(int, input().split())
a = list(map(int, input().split()))

# Call the function and print the result
print(snuke_possible(n, k, a))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03651/p03651_num6_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

bool isPossible(vector<int>& balls, int K) {
    int N = balls.size();
    int maxBall = 0;
    
    for (int i = 0; i < N; i++) {
        maxBall = max(maxBall, balls[i]);
    }
    
    if (K > maxBall || K % 2 != maxBall % 2) {
        return false;
    }
    
    return true;
}

int main() {
    int N, K;
    cin >> N >> K;
    
    vector<int> balls(N);
    
    for (int i = 0; i < N; i++) {
        cin >> balls[i];
    }
    
    if (isPossible(balls, K)) {
        cout << "POSSIBLE" << endl;
    } else {
        cout << "IMPOSSIBLE" << endl;
    }
    
    return 0;
}